src/ai/gate_jirwin.py

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env python
+
+import rospy
+from robosub.msg import visionPos as vision_pos
+from robosub.msg import control
+
+
+class Node():
+    def __init__(self):
+        self.pub = rospy.Publisher('control', control, queue_size=1)
+        self.sub = rospy.Subscriber('vision/buoy', vision_pos, self.callback)
```

Comment:
Your subscribers seem backwards. The buoy processing is subscribed to the gate topic and the gate processing to the buoy topic